pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -996,6 +1033,17 @@ public void addUnAckedMessages(int numberOfMessages) {
         topic.getBrokerService().addUnAckedMessages(this, numberOfMessages);
     }
 
+    @Override
+    public synchronized void afterAckMessages(Throwable exOfDeletion, Object ctxOfDeletion) {
```

Comment:
This is a hot path in Pulsar. Is it better to use compareAndSet() with AtomicBoolean to avoid any performance degradation?

And it's better to remove `volatile` since you already have `synchronized` here.

Reply:
yes，agree to use `AtomicIntegerFieldUpdater ` to update `blockedDispatcherOnCursorDataCanNotFullyPersist`

Reply:
I created a stat class `BlockDispatcherSignatureOnCursorDataCanNotFullyPersist` to avoid use `synchronized `, please take a look again.